front/app/components/IdeaCards/shared/Filters/SortingBox/index.tsx

```diff
@@ -24,8 +26,9 @@ const SortingBox = ({ handleSortOnChange, phaseId }: SortingBoxProps) => {
   const phaseDefaultSort = phase?.data.attributes.ideas_order;
 
   const [searchParams] = useSearchParams();
-  const currentSortType =
-    searchParams.get('sort') || phaseDefaultSort || 'trending';
+  const currentSortType = (searchParams.get('sort') ||
+    phaseDefaultSort ||
+    'trending') as IdeaSortMethod | null;
```

Comment:
```suggestion
  const searchParamsSort = searchParams.get('sort') as IdeaSortMethod | null;
  const currentSortType =
    searchParamsSort || phaseDefaultSort || 'trending';
```
To reduce the chance of silent bugs. Type casting has introduced many in the past, so we should be careful with it. Let's apply it to just the problematic part. (And this is another reason to look into e.g. context to keep the filters, so we don't have to rely on type casting.)

If you go with this, an error re: unnecessary condition will pop up as well.

Reply:
Good call :+1: